app/services/telegram/incoming_message_service.rb

```diff
@@ -100,6 +100,9 @@ def file_content_type
   def attach_files
     return unless file
 
+    file_download_path = inbox.channel.get_telegram_file_path(file[:file_id])
```

Comment:
WDYT of using a function to check if the URL is valid?

```rb
def valid_url?(url)
  return false if url.nil? || url.strip.empty?

  uri = URI.parse(url)
  %w[http https].include?(uri.scheme) && !uri.host.nil?
rescue URI::InvalidURIError
  false
end
```

Reply:
The download URL is always valid here. We are handling the case when the download response is empty for some reason.